src/main/java/info/solidsoft/mockito/java8/AssertionMatcher.java

```diff
@@ -62,7 +60,7 @@ public boolean matches(T argument) {
         try {
             consumer.accept(argument);
             return true;
-        } catch (AssertionError e) {
+        } catch (AssertionError | Exception e) {
```

Comment:
This implementation hides any random exception which occurs in the lamba. The exception has to be caught and rethrown wrapped with a runtime exception.